### PR TITLE
Request to merge Milestone-4-filter branch

### PR DESCRIPTION
**Pull request by mikenath223**
Implementations:
* We enhanced the redux store by creating a filter reducer to update the value of filter in the Redux store.
* We also implemented the category filter component which renders a select field with filter categories.
* Finally we kept the filter value in the Redux store, by implementing a handleFilterChange method which modifies the filter state in the Redux store and passed it down as a prop to CategoryFilter component.

Kindly review.

Thanks.

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const filterCategory = (state = '', action) => {
+  switch (action.type) {
+    case FILTER_BOOK:
+      return action.category;
+    default:
+      return state;
+  }
+};
+
```
Comment (shloch) [inline src/reducers/filter.js]: it should default to `"all"`according to [documentation](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/ItDefaultsToAll.png) not empty string

File: src/containers/category_filter.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
```
Comment (shloch) [inline src/containers/category_filter.js]: `category_filter.js` is a "presentational" component, and according to this project's specification, presentational-components should be in the folder `"src/components/"` folder.

### please relocate this file

**Review by shloch — APPROVED** [review]
## hello @mikenath223 , @ViriCruz 

Nice execution.Nothing much to say here, you surely know what you're doing.
The project is in line with specification. I have 2 minor observations below to completely align with the specification (especially with the file to relocate below)

### Please handle this little thing prior to last thumbs up...Happy coding !!

File: src/containers/category_filter.js (hunk 2 above)
Comment (mikenath223) [reply src/containers/category_filter.js]: All done.👍

**Review by azdrenymeri — APPROVED** [review]
#### Requirements are met :heavy_check_mark: 
#### You can merge it :+1:

**Comment by mikenath223** [comment]
Thanks a lot for the review. We will merge now.

**Pull request merged**